soc/realtek/ec/rts5912/poweroff.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2025 Realtek Semiconductor Corporation, SIBG-SD7
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/poweroff.h>
+#include <zephyr/toolchain.h>
+#include "rts5912_ulpm.h"
+
+void z_sys_poweroff(void)
+{
+#ifdef CONFIG_SOC_RTS5912_ULPM
+	rts5912_ulpm_enable();
+#endif /* CONFIG_SOC_RTS5912_ULPM */
```

Comment:
It's preferred to use `IS_ENABLED` instead of `#ifdef` when possible. 

```suggestion
	if (IS_ENABLED(CONFIG_SOC_RTS5912_ULPM) {
		rts5912_ulpm_enable();
	}
```

Reply:
done.